helm/cluster-azure/templates/_azure_cluster.tpl

```diff
@@ -21,14 +21,54 @@ spec:
         - {{ .Values.connectivity.network.controlPlane.cidr }}
       - name: node-subnet
         natGateway:
-          name: node-natgateway
+          name: {{ include "resource.default.name" $ }}-node-natgateway
+          NatGatewayIP:
+            name: {{ include "resource.default.name" $ }}-node-natgateway-ip
         role: node
         cidrBlocks:
         - {{ .Values.connectivity.network.workers.cidr }}
     vnet:
       name: {{ include "resource.default.name" $ }}-vnet
       cidrBlocks:
       - {{ .Values.connectivity.network.hostCidr }}
+      {{- if .Values.providerSpecific.network.peerings }}
```

Comment:
should this be in the `connectivity` section or is in `providerSpecific` on purpose ?